src/main/java/org/candlepin/subscriptions/product/OfferingSyncController.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.product;
+
+import java.util.Objects;
+import java.util.Optional;
+import org.candlepin.subscriptions.db.OfferingRepository;
+import org.candlepin.subscriptions.db.model.Offering;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/** Update {@link Offering}s from product service responses. */
+@Component
+public class OfferingSyncController {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OfferingSyncController.class);
+
+  private final OfferingRepository offeringRepository;
+
+  public OfferingSyncController(OfferingRepository offeringRepository) {
+    this.offeringRepository = offeringRepository;
+  }
+
+  /**
+   * Persists the latest state of an Offering. If no stored Offering matches the SKU, then the
+   * Offering is inserted into the datastore. Otherwise, if there are actual changes, the stored
+   * Offering with the matching SKU is updated with the given Offering.
+   *
+   * @param newState the updated Offering
+   */
+  public void syncOffering(Offering newState) {
+    String sku = newState.getSku();
+    if (newState.getProductIds() == null || newState.getProductIds().isEmpty()) {
+      LOGGER.warn("The given sku=\"{}\" contained no engineering products. Skipping sync.", sku);
+      return;
+    }
```

Comment:
We actually do care about SKUs not providing content. I had to go look at the MW01484 to be sure. So maybe a better filter would just be against `PRODUCT_CATEGORY`?

Reply:
I thought there was a requirement on this to deny syncing offerings that had no engprod OIDs, but such a requirement is not in ENT-2658. It turns out that the requirement was in ENT-2664: "Validation: we should log a warning for products with no engineering product IDs (and not store any information)." Since there is an offering with no engProds, should I remove this check entirely?

Reply:
Just to reiterate an earlier conversation for visibility, let's implement as originally scoped, and we'll revisit with a tweak after the whole feature is done.